sui_programmability/examples/defi/sources/pool.move

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Example implementation of a liquidity Pool for Sui.
+///
+/// - Only module publisher can create new Pools.
+/// - For simplicity's sake all swaps are done with SUI coin.
```

Comment:
```suggestion
/// - For simplicity's sake all swaps are done with SUI coin.
/// Generalizing to swaps between non-SUI coin types requires a few more generics, but is otherwise straightforward
```